src/Adapter/Module/CommandHandler/ResetModuleHandler.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Module\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsCommandHandler;
+use PrestaShop\PrestaShop\Core\Domain\Module\Command\ResetModuleCommand;
+use PrestaShop\PrestaShop\Core\Domain\Module\CommandHandler\ResetModuleHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Module\Exception\CannotResetModuleException;
+use PrestaShop\PrestaShop\Core\Module\ModuleManager;
+use PrestaShop\PrestaShop\Core\Module\ModuleRepository;
+
+#[AsCommandHandler]
+class ResetModuleStatusHandler implements ResetModuleHandlerInterface
+{
+    public function __construct(
+        protected ModuleManager $moduleManager,
+        protected ModuleRepository $moduleRepository,
+    ) {
+    }
+
+    public function handle(ResetModuleCommand $command): void
+    {
+        if (!$this->moduleManager->isInstalled($command->getTechnicalName()->getValue())) {
```

Comment:
In the `GetModuleInfosHandler` we added a check to be sure the module is present, if not we return a 404 I think we should homogenize this behaviour in all actions So that if someone tries to do an action on a non-existent module he will always get a 404

To help doing that everywhere can you please move this code from the get handler:
```php
$module = $this->moduleRepository->getModule($query->getTechnicalName()->getValue());
        if (!$module->disk->get('is_present')) {
            throw new ModuleNotFoundException('Cannot find Module with name ' . $query->getTechnicalName()->getValue());
        }
```

into a `ModuleRepository::getPresentModule` method that performs this check and returns the `$module` at the end. This way n all handlers (including `GetModuleInfosHandler` and this handler) we can always start by doing:
```php
    $module = $this->moduleRepository->getPresentModule($technicalName);
```
and start working from this. This way we'll guaranty the not found use case and the 404 is correctly handled everywhere. And you can even use the returned instance to check if it's enabled or installed based on its attributes.